System/Constants.h

```diff
@@ -3,11 +3,31 @@
 
 namespace RTE {
 
+#pragma region Type Definitions
+	typedef int MOID; //!< Distinctive type definition for MovableObject IDs.
```

Comment:
Wut why? This is silly and will just make people think MOID is a special thing when it's just an int. I'm against this!

Reply:
I don't know why but I find this a silly argument. Alot of things are just ints yet they are explicitly defined as "special".
Making this just int will drag alot of changes and will make it less obvious when dealing with IDs.
I think it's great that this is branded as not-just-an-int.

Reply:
What other things are just ints but defined as special?
The bunch of changes is a fair point, but it's not like you're not making a bunch of changes anyway lol. The making it less obvious thing I pretty much disagree with, though it depends to some degree on use-cases and code design, wherein it's only a problem if your code is ass (which, granted, a lot of CC is still).
If you have a method `SetID(int MOIDToSet) {...}` it's no less clear what it does than `SetID(MOID MOIDToSet) {...}` and it makes it clear what type MOIDToSet is instead of obfuscating it.
Similarly, if you were getting the MOID, the only way it'd be unclear what you're working with is if you used stupid terrible variable names.
```
int a = obj.GetMOID(); //What is a? Who knows, it's got a bad name
MOID b = obj.GetMOID(); //I guess you know it's an MOID, but it doesn't really help much anyway does it?
```
vs.
```
int objectMOID = obj.GetMOID(); //The variable name makes it 100% clear that you're working with an MOID, you don't need to have a special type here.
MOID objectMOID2 = obj.GetMOID(); //This is no clearer by having MOID type, and it, again, makes it unclear that you're actually working with an int.
```

Reply:
In a sense, a MOID represents an entity so making it just a number adds to the confusion.
I don't know, maybe i'm not looking at it correctly because I didn't dig too much into it.
Anyway this is outside the scope of this PR, feel free to do this in a new one because i'm really not bothered by this, at least at the moment.

Reply:
Was MOID used instead of int before and you just pulled it into constants? I'd thought you'd made it.

Reply:
I didn't, both of MOID and MID were defined in FrameMan.